### PR TITLE
MAINT: refine error message for __array_ufunc__ not implemented

### LINKED ISSUE #9079: Misleading __array_ufunc__ error
Here's a very simple `__array_ufunc__`, which is likely to be one of the first tests people try:
```python
class foo(np.ndarray):
     def __array_ufunc__(self, *args, **kwargs):
         print(args, kwargs)
         return super().__array_ufunc__(*args, **kwargs)
```

Unfortunately, the error given is a little unhelpful:

```
>>> a = np.array(1).view(foo)
>>> a + a
(<ufunc 'add'>, '__call__', foo(1), foo(1)) {}
(<ufunc 'add'>, '__call__', foo(1), foo(1)) {}
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-24070be0e9db> in <module>()
----> 1 a + a

TypeError: operand type(s) do not implement __array_ufunc__(<ufunc 'add'>, '__call__', foo(1), foo(1)): 'foo', 'foo'
```

Some kind of hint that `ndarray.__array_ufunc__` was not able to handle things either might be helpful.

### PR DESCRIPTION
**Pull request by shoyer**
xref #9079

@charris can we squeeze this in for 1.13? This seems like a meaningful usability enhancement for this new feature

### CODE REVIEW

File: numpy/core/_internal.py [hunk 1]
```diff
@@ -695,8 +695,8 @@ def array_ufunc_errmsg_formatter(dummy, ufunc, method, *inputs, **kwargs):
                              for k, v in kwargs.items()])
     args = inputs + kwargs.get('out', ())
     types_string = ', '.join(repr(type(arg).__name__) for arg in args)
-    return ('operand type(s) do not implement __array_ufunc__'
-            '({!r}, {!r}, {}): {}'
+    return ('operand type(s) all returned NotImplemented upon calling '
```
Comment (eric-wieser) [inline numpy/core/_internal.py]: "from" would be shorter than "upon calling", but either is fine

**Review by eric-wieser — APPROVED** [review]
LGTM. Hopefully not too late for 1.13!

File: numpy/core/_internal.py (hunk 1 above)
Comment (shoyer) [inline numpy/core/_internal.py]: done

**Comment by charris** [comment]
1.13 is branched. Might could backport this, there seems to be at least one more fixup that need to be made.

**Comment by juliantaylor** [comment]
note that if you base it on the merge base between master and 1.13 you can just merge the same commit in both branches

**Comment by charris** [comment]
I'd label this maintenance, enhancements are for new things, this is more an improvement of an existing function.

**Comment by shoyer** [comment]
OK, edited commit name from ENH -> MNT.

For the backport, I just open up another PR with the same branch targeting `maintanence/1.13.x`?

**Comment by charris** [comment]
Thanks Stephan.

**Pull request merged**